src/components/common/ThemeToggle/index.tsx

```diff
@@ -0,0 +1,98 @@
+import MonitorIcon from '@/public/assets/icons/monitor.svg';
+import MoonIcon from '@/public/assets/icons/moon.svg';
+import SunIcon from '@/public/assets/icons/sun.svg';
+import { useTheme } from 'next-themes';
+import { useEffect, useId, useState } from 'react';
+import styles from './style.module.scss';
+
+const ThemeToggle = () => {
+  const { theme, setTheme } = useTheme();
+  const [mounted, setMounted] = useState(false);
+
+  const lightId = `light${useId()}`;
+  const systemId = `system${useId()}`;
+  const darkId = `dark${useId()}`;
+
+  const iconSize = 30;
+
+  const calcIndicatorStyle = (newTheme: string | undefined) => {
```

Comment:
Does this type check? `styles` is not a record of strings because of the style.module.scss.d.ts, and `useTheme`'s `theme` returns `string | undefined`, so you'd have to check if it's equal to one of the class names as a literal before being able to do `styles[theme]`.